hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/analysis/HoodieAnalysis.scala

```diff
@@ -428,6 +428,12 @@ case class ResolveImplementationsEarly() extends Rule[LogicalPlan] {
         if sparkAdapter.isHoodieTable(table) && ct.query.forall(_.resolved) =>
         CreateHoodieTableAsSelectCommand(table, mode, query)
 
+      case ct: CreateTable =>
+        // NOTE: In case of CreateTable with schema and multiple partition fields,
+        // we have to make sure that partition fields are ordered in the same way as they are in the schema.
+        HoodieSchemaUtils.checkPartitionSchemaOrder(ct.tableDesc.schema, ct.tableDesc.partitionColumnNames)
```

Comment:
like `df.saveAsTable` or `create table as select`, the tableDesc will not have schema, may be need use query logical plan output

Reply:
CTAS is handled in the case above but good point about `df.saveAsTable`. Will fix it.

Reply:
Fixed.